File: discord_bot/src/main/java/the/kis/devs/discordbot/permission/IPermission.java
package the.kis.devs.discordbot.permission;

import net.dv8tion.jda.api.entities.Member;
import the.kis.devs.discordbot.util.Globals;

/**
 * @author _kisman_
 * @since 14:52 of 23.06.2022
 */
public interface IPermission extends Globals {
    boolean valid(Member member);
}

File: server/src/main/kotlin/the/kis/devs/server/permission/IPermission.kt
package the.kis.devs.server.permission

import me.yailya.sockets.server.SocketServerConnection

/**
 * @author _kisman_
 * @since 22:03 of 05.07.2022
 */
interface IPermission {
    fun check(connection : SocketServerConnection) : Boolean
}